modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/spds/aah.cc

```diff
@@ -50,15 +50,38 @@ AAH_SPDS::AAH_SPDS(int id, const Vector3& omega, const MeshContinuum& grid, bool
   }
 
   // Generate topological ordering
-  spls_.item_id.clear();
-  boost::topological_sort(local_cell_graph, std::back_inserter(spls_.item_id));
-  std::reverse(spls_.item_id.begin(), spls_.item_id.end());
-  if (spls_.item_id.empty())
+  spls_.clear();
+  boost::topological_sort(local_cell_graph, std::back_inserter(spls_));
+  std::reverse(spls_.begin(), spls_.end());
+  if (spls_.empty())
   {
     throw std::logic_error("AAH_SPDS: Cyclic dependencies found in the local cell graph.\n"
                            "Cycles need to be allowed by the calling application.");
   }
 
+  // Generate levelized spls
+  int max_level = 0;
+  std::vector<int> levels(num_vertices(local_cell_graph), 0);
+  for (auto it = spls_.begin(); it != spls_.end(); ++it)
```

Comment:
I noticed, that you could do: `for (auto & v: spls_) {...` now. And you don't need the `auto v = *it;` any more.

Reply:
Fixed.